libconcurrent/concurrent/hsynch.c

```diff
@@ -69,11 +69,18 @@ void HSynchThreadStateInit(HSynchStruct *l, HSynchThreadState *st_thread, int pi
         }
     } else {
         int ncpus = numa_num_configured_cpus();
-        if (numa_node_of_cpu(0) == numa_node_of_cpu(ncpus / 2)) {
-            if (getPreferedCore() >= ncpus / 2)
-                node_of_thread = (getPreferedCore() - ncpus / 2) / (l->numa_node_size / 2);
-            else
-                node_of_thread = getPreferedCore() / (l->numa_node_size / 2);
+        if (numa_node_of_cpu(0) == numa_node_of_cpu(ncpus / 2) && ncpus > 1) {
+            int manual_numa_node;
```

Comment:
You can omit this declaration and use directly `manual_numa_node`

Reply:
done